### PR TITLE
Dev python/spyder: version bump 4.1.3 and deps

### PR DESCRIPTION
**Pull request by AndrewAmmerlaan**
Version bump of spyder and some deps

also bumped flake8 and friends as this is required by the new python-language-server

### CODE REVIEW

File: dev-python/python-language-server/python-language-server-0.32.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{6,7} )
```
Comment (mattst88) [inline dev-python/python-language-server/python-language-server-0.32.0.ebuild:6]: Can we add Python 3.8 support?

File: dev-python/pyflakes/pyflakes-2.2.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+DESCRIPTION="Passive checker for Python programs"
+HOMEPAGE="https://github.com/PyCQA/pyflakes https://pypi.org/project/pyflakes/"
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~mips ~ppc ~ppc64 ~s390 ~sparc ~x86 ~amd64-linux ~x86-linux ~ppc-macos ~x64-macos ~x86-macos ~sparc-solaris ~sparc64-solaris ~x64-solaris ~x86-solaris"
+
+RDEPEND="${BDEPEND}"
```
Comment (mattst88) [inline dev-python/pyflakes/pyflakes-2.2.0.ebuild:17]: pyflake-2.1.1.ebuild has

```
BDEPEND="dev-python/setuptools[${PYTHON_USEDEP}]"
RDEPEND="${BDEPEND}"
```

Should this new version also have that? Just wondering whether the BDEPEND=... was dropped intentionally.

File: dev-python/sphinx-prompt/metadata.xml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+  <maintainer type="project">
+    <email>python@gentoo.org</email>
```
Comment (mattst88) [inline dev-python/sphinx-prompt/metadata.xml:5]: @gentoo/python should sign-off on new packages under their maintainership.

Also, I think the commit title should say something like 'dev-python/sphinx-prompt: Add new package'.

File: dev-python/sphinx-prompt/metadata.xml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+    <name>Python</name>
+  </maintainer>
+  <upstream>
+    <remote-id type="pypi">flake8</remote-id>
+    <remote-id type="gitlab">pycqa/flake8</remote-id>
```
Comment (mattst88) [inline dev-python/sphinx-prompt/metadata.xml:10]: These two lines look like a copy and paste mistake.

File: dev-python/spyder-kernels/spyder-kernels-1.9.1.ebuild [hunk 5]
```diff
@@ -0,0 +1,45 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{6,7} )
```
Comment (mattst88) [inline dev-python/spyder-kernels/spyder-kernels-1.9.1.ebuild:6]: Is it possible to add Python 3.8 support?

File: dev-python/spyder/spyder-4.1.3.ebuild [hunk 6]
```diff
@@ -0,0 +1,156 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{6,7} )
```
Comment (mattst88) [inline dev-python/spyder/spyder-4.1.3.ebuild:6]: Python 3.8?

File: dev-python/spyder-vim/spyder-vim-0_pre20200514.ebuild [hunk 7]
```diff
@@ -0,0 +1,34 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{6,7} )
```
Comment (mattst88) [inline dev-python/spyder-vim/spyder-vim-0_pre20200514.ebuild:6]: Python 3.8?

**Comment by mattst88** [comment]
Thanks! I left a few comments. To be clear, the questions about whether it's possible to add Python 3.8 can definitely be answered with "no, not yet, there are too many dependent packages that need Python 3.8 support first". I just want to make sure that that's actually the case, and that we're not just leaving out 3.8 support for no reason.

File: dev-python/python-language-server/python-language-server-0.32.0.ebuild (hunk 1 above)
Comment (AndrewAmmerlaan) [reply dev-python/python-language-server/python-language-server-0.32.0.ebuild:6]: Nope, sadly not, dependencies were missing py3_8 support last I checked

File: dev-python/pyflakes/pyflakes-2.2.0.ebuild (hunk 2 above)
Comment (AndrewAmmerlaan) [reply dev-python/pyflakes/pyflakes-2.2.0.ebuild:17]: Dependency on setuptools is automatically added by distutils eclass

File: dev-python/sphinx-prompt/metadata.xml (hunk 3 above)
Comment (AndrewAmmerlaan) [reply dev-python/sphinx-prompt/metadata.xml:5]: I added this package because it is a dependency for building flake8's documentation:

```
# Sphinx and friends are not keyworded for all arches, repoman complains
#distutils_enable_sphinx docs/source dev-python/sphinx-prompt dev-python/sphinx_rtd_theme
```

While the package works, repoman is unhappy that sphinx and other deps do not have the same keywords as flake8. So, for flake8 I either have to drop arches, or leave out the doc support. (or restricting the usage of the doc flag to certain arches, but I tried that once and it gets messy)

I figured that even if I can't add doc support to flake8 at the moment, I could still add the sphinx-prompt package in case it becomes possible to add this doc support later. (in case sphinx gains some keywords)

As flake8 is maintained by the python team, I added them to sphinx-prompt metadata as well. As far as I know, the python team allows that they are added as maintainer to packages that are dependencies of packages they already maintain. Though it would be great if someone from the python team could confirm this.

File: dev-python/sphinx-prompt/metadata.xml (hunk 4 above)
Comment (AndrewAmmerlaan) [reply dev-python/sphinx-prompt/metadata.xml:10]: Correct, thanks, will fix it.

File: dev-python/spyder-kernels/spyder-kernels-1.9.1.ebuild (hunk 5 above)
Comment (AndrewAmmerlaan) [reply dev-python/spyder-kernels/spyder-kernels-1.9.1.ebuild:6]: Same, as pyls. To be more specific I need py3_8 support on jupyter and friends maintained by the science team.

It would be great to add py3_8 to all these ebuilds, especially considering spyder officially supports py3_8 from this release onwards. But, I'm not going to make a huge PR that adds py3_8 to all the dependencies that need it. Such a large PR will just get ignored, or it will take very long to be reviewed, and by that time the science team will likely have added py3_8 to their ebuilds anyway.

Tl;dr Yes it is possible but it will make the PR huge, and therefore will likely not be worth the effort.

File: dev-python/spyder/spyder-4.1.3.ebuild (hunk 6 above)
Comment (AndrewAmmerlaan) [reply dev-python/spyder/spyder-4.1.3.ebuild:6]: Same as above, can't add py3_8 because I need py3_8 on pyls and spyder-kernels first (and some other deps, ipython and such)

File: dev-python/spyder-vim/spyder-vim-0_pre20200514.ebuild (hunk 7 above)
Comment (AndrewAmmerlaan) [reply dev-python/spyder-vim/spyder-vim-0_pre20200514.ebuild:6]: Need py3_8 on spyder

**Comment by AndrewAmmerlaan** [comment]
> Thanks! I left a few comments. To be clear, the questions about whether it's possible to add Python 3.8 can definitely be answered with "no, not yet, there are too many dependent packages that need Python 3.8 support first". I just want to make sure that that's actually the case, and that we're not just leaving out 3.8 support for no reason.

Thanks, I fixed up sphinx-prompt's metadata file, I tend to copy metadata files and then completely forget about them. The py3_8 support is something I really would like to add,  especially considering that spyder runs the programs written in it with the same python version as it is running in (without configuration). Meaning that no py3_8 support in the ebuild makes it more difficult to execute the programs that you write with spyder with python3_8. But I'm still waiting on the science team to add py3_8 support to jupyter/ipython and friends. (Though I really had hoped I would be able to add py3_8 to the ebuild in this version bump)

tl;dr py3_8 is possible, but not without touching a great deal of packages maintained by the science team.

File: dev-python/flake8/flake8-3.8.1.ebuild [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+	>=dev-python/pyflakes-2.2.0[${PYTHON_USEDEP}]
+	<dev-python/pyflakes-2.3.0[${PYTHON_USEDEP}]
+	>=dev-python/pycodestyle-2.6.0[${PYTHON_USEDEP}]
+	<dev-python/pycodestyle-2.7.0[${PYTHON_USEDEP}]
+	$(python_gen_cond_dep '
+		dev-python/configparser[${PYTHON_USEDEP}]
+		dev-python/enum34[${PYTHON_USEDEP}]
+		dev-python/functools32[${PYTHON_USEDEP}]
+		dev-python/typing[${PYTHON_USEDEP}]
+	' -2)
```
Comment (a17r) [inline dev-python/flake8/flake8-3.8.1.ebuild:32]: None of those make sense.

**Comment by Zlogene** [comment]
general note: please remove py36 from compat, py37 is now default and we want to ditch py36 ASAP.

File: dev-python/flake8/flake8-3.8.1.ebuild (hunk 8 above)
Comment (AndrewAmmerlaan) [reply dev-python/flake8/flake8-3.8.1.ebuild:32]: This is copied from the previous ebuild. Looking at the setup.cfg file it looks like these are only required for older python versions: https://github.com/PyCQA/flake8/blob/master/setup.cfg

I'll fix it

Comment (AndrewAmmerlaan) [reply dev-python/flake8/flake8-3.8.1.ebuild]: Fixed

**Comment by AndrewAmmerlaan** [comment]
> general note: please remove py36 from compat, py37 is now default and we want to ditch py36 ASAP.

Dropped as far as possible without making this PR even bigger then it already is.

**Comment by AndrewAmmerlaan** [comment]
@juippis @mattst88 @a17r If everything is now fine, could someone please merge this?

I think I addressed everyone's comments and suggestions, please let me know if I missed anything.

**Comment by mattst88** [comment]
Thanks again!

**Comment by AndrewAmmerlaan** [comment]
> Thanks again!

No problem, Thank you for the review and merge :)
